slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java

```diff
@@ -0,0 +1,148 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.interchange;
+
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestationsDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlocksDao;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+import tech.pegasys.web3signer.slashingprotection.dao.ValidatorsDao;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.Metadata;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+
+public class InterchangeV5Exporter {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int FORMAT_VERSION = 5;
+
+  private final Jdbi jdbi;
+  private final ValidatorsDao validatorsDao;
+  private final SignedBlocksDao signedBlocksDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+  private final ObjectMapper mapper;
+
+  public InterchangeV5Exporter(
+      Jdbi jdbi,
+      ValidatorsDao validatorsDao,
+      SignedBlocksDao signedBlocksDao,
+      SignedAttestationsDao signedAttestationsDao,
+      ObjectMapper mapper) {
+    this.jdbi = jdbi;
+    this.validatorsDao = validatorsDao;
+    this.signedBlocksDao = signedBlocksDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+    this.mapper = mapper;
+  }
+
+  public void export(final OutputStream out) throws IOException {
+    try (final JsonGenerator jsonGenerator = mapper.getFactory().createGenerator(out)) {
+      jsonGenerator.writeStartObject();
+
+      final Metadata metadata = new Metadata(FORMAT_VERSION, Bytes.fromHexString("FFFFFFFF"));
+
+      jsonGenerator.writeFieldName("metadata");
+      mapper.writeValue(jsonGenerator, metadata);
+
+      jsonGenerator.writeArrayFieldStart("data");
+      populateInterchangeData(jsonGenerator);
+      jsonGenerator.writeEndArray();
+
+      jsonGenerator.writeEndObject();
+    }
+  }
+
+  private void populateInterchangeData(final JsonGenerator jsonGenerator) {
+    jdbi.useTransaction(
+        h ->
+            validatorsDao
+                .findAllValidators(h)
+                .forEach(
+                    validator -> {
+                      LOG.info(
+                          "Exporting entries for validator {}",
+                          validator.getPublicKey().toHexString());
+                      try {
+                        populateValidatorRecord(h, validator, jsonGenerator);
+                      } catch (IOException e) {
```

Comment:
nit: final ... and you can also raise `UncheckedIOException` which wraps IOException neatly.

Reply:
done